docker-amundsen-nebula.yml

```diff
@@ -0,0 +1,353 @@
+version: '3'
+services:
+  elasticsearch:
+      image: elasticsearch:8.0.0
+      container_name: es_amundsen
+      ports:
+          - 9200:9200
+      networks:
+        - amundsennet
+      ulimits:
+        nofile:
+          soft: 65536
+          hard: 65536
+      environment:
+        - discovery.type=single-node
+        - xpack.security.enabled=false
+  amundsensearch:
+      build:
+        context: .
+        dockerfile: Dockerfile.search.public
+      container_name: amundsensearch
+      ports:
+        - 5001:5001
+      depends_on:
+        - elasticsearch
+      networks:
+        - amundsennet
+      environment:
+        - PROXY_ENDPOINT=es_amundsen
+  amundsenmetadata:
+      build:
+        context: .
+        dockerfile: Dockerfile.metadata.public
+      container_name: amundsenmetadata
+      depends_on:
+        - graphd
+      ports:
+        - 5002:5000
+      networks:
+        - amundsennet
+      environment:
+         - PROXY_HOST=graphd
+         - PROXY_PORT=9669
+         - PROXY_CLIENT=NEBULA
+         - CREDENTIALS_PROXY_USER=root
+         - CREDENTIALS_PROXY_PASSWORD=nebula
+         - PROXY_ENCRYPTED=False
+  amundsenfrontend:
+      build:
+        args:
+          SEARCHSERVICE_BASE: http://amundsensearch:5001
+          METADATASERVICE_BASE: http://amundsenmetadata:5002
+        context: .
+        dockerfile: Dockerfile.frontend.local
+      container_name: amundsenfrontend
+      depends_on:
+        - amundsenmetadata
+        - amundsensearch
+      ports:
+        - 5000:5000
+      networks:
+        - amundsennet
+      environment:
+        - SEARCHSERVICE_BASE=http://amundsensearch:5001
+        - METADATASERVICE_BASE=http://amundsenmetadata:5002
+        - FRONTEND_SVC_CONFIG_MODULE_CLASS=amundsen_application.config.TestConfig
+
+  metad0:
+    # Will use switch to a versioned image tag when nebula releases 3.1.0 in upcoming weeks
+    image: vesoft/nebula-metad@sha256:30b421feff7845fdd15dea4a97e89ce9beeb243cf984b63609c64cd754137cf3
```

Comment:
Nebula Graph 3.1.0 will be released in the upcoming weeks, afterwards, I will update this to a tag.